subprojects/docs/src/docs/userguide/javaLibraryPlugin.xml

```diff
@@ -0,0 +1,276 @@
+        <sample id="useJavaLibraryPlugin" dir="java-library/quickstart" title="A dependency type is used in the public API">
+            <sourcefile file="src/main/java/org/gradle/HttpClientWrapper.java" snippet="api-dependency"/>
+        </sample>
+        <para>On the other hand, the <literal>ExceptionUtils</literal> type, coming from the <literal>commons-lang</literal> library, is only used in a method
+            body, so it's an implementation dependency:
+        </para>
+        <sample id="useJavaLibraryPlugin" dir="java-library/quickstart" title="A dependency type is used in the method bodies only">
+            <sourcefile file="src/main/java/org/gradle/HttpClientWrapper.java" snippet="implementation-dependency"/>
+        </sample>
+        <para>Therefore, we can deduce that <literal>commons-httpclient</literal> is an API dependency, whereas <literal>commons-lang</literal> is an implementation
+            dependency, which directly translates into the build file:
+        </para>
+        <sample id="useJavaLibraryPlugin" dir="java-library/quickstart" title="Declaring API and implementation dependencies">
+            <sourcefile file="build.gradle" snippet="dependencies"/>
+        </sample>
+        <para>It is in general safer to assume that a dependency is an implementation dependency only: leakage of implementation types to consumers would
+            then directly lead to a compile error, which would be solved either by removing the type from the public API, or promoting the dependency as an
+            API dependency instead.
+        </para>
+    </section>
+
+    <section id="sec:java_library_configurations_graph">
+        <title>The Java Library plugin configurations</title>
+        <para>The following graph describes the configuration setup when the Java Library plugin is in use. The configurations in <firstterm>grey</firstterm> are
+            deprecated and only exist for backwards compatibility.
+        </para>
+        <imageobject>
+            <imagedata fileref="img/java-library-configurations.png"/>
+        </imageobject>
+        <para>If we exclude the deprecated configurations, the next graph shows what you should care about as a user:</para>
+        <imageobject>
+            <imagedata fileref="img/java-library-ignore-deprecated.png"/>
+        </imageobject>
+        <itemizedlist>
+            <listitem>The configurations in <firstterm>green</firstterm> are the ones a user should use to declare dependencies
+            </listitem>
+            <listitem>The configurations in <firstterm>pink</firstterm> are the ones used when a component compiles, or runs against the library
+            </listitem>
+            <listitem>The configurations in <firstterm>blue</firstterm> are internal to the component, for its own use
+            </listitem>
+            <listitem>The configurations in <firstterm>white</firstterm> are legacy configurations that can be used to declare dependencies too
```

Comment:
What does "legacy" mean? To me it sounds negative, as if they will disappear (which is not the case)